### PR TITLE
OKTA-508061 :- adding new KEY in SIW for EML SSU success message

### PR DESCRIPTION
**Pull request by vicentevillegas-okta**
## Description:
This change is for adding a new KEY to the login.properties file, so we can personalize a view message including the Application name. This is a follow up work for this [PR](https://github.com/okta/okta-core/pull/67284)

This is the [Figma link](https://www.figma.com/file/fZGJS3FV0RpfKImY8CKwij/Magic-Link?node-id=2009%3A57303)

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [x] Added e2e tests

- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

- Did you verify the change by running downstream monolith artifacts? (YES | NO | UNSURE)

### Screenshot/Video:
[Unlock with landing directly to app](https://okta.box.com/s/sva1lyotlx7ohuyp5ttjum32eotn6gxj)
[Unlock showing list of auth](https://okta.box.com/s/798n8ciy0ysdlgqqwkec4gzpdlilfn10)
[Unlock showing list of auth - rain](https://okta.box.com/s/dbctxrlgi4c8cb2mb98rgsyzi65fqm24)

image

image

### Reviewers:
@justinabrokwah-okta @kumarabhinav-okta @IanHakes-okta @okta/ciamx 

### Issue:
- [OKTA-508061](https://oktainc.atlassian.net/browse/OKTA-508061)

### Bacon
[Green bacon](https://bacon-go.aue1e.saasure.net/commits?artifact=okta-signin-widget&branch=VV-OKTA-508061&page=1&pageSize=1&sha=53d5e2e96dd0c248ceaf4d5ced36b8a59f049dde&tab=main)

image

### CODE REVIEW

File: packages/@okta/i18n/src/properties/login.properties [hunk 1]
```diff
@@ -1276,6 +1276,7 @@ oie.user.profile.division = Division
 oie.user.profile.department = Department
 oie.user.profile.managerId = Manager ID
 oie.user.profile.manager = Manager
+oie.selfservice.unlock_user.success.message.with.application.name = Account successfully unlocked!<br>Continue to access {0}.
```
Comment (vicentevillegas-okta) [inline packages/@okta/i18n/src/properties/login.properties:1279]: @IanHakes-okta could you please take a look to this? here is the [figma](https://www.figma.com/file/fZGJS3FV0RpfKImY8CKwij/Magic-Link?node-id=2009%3A57303)

**Review by justinabrokwah-okta — COMMENTED** [review]
PR looks good! Already discussed offline about the UI

File: test/testcafe/spec/UnlockAccountChallenge_spec.js [hunk 2]
```diff
@@ -117,3 +129,28 @@ test.requestHooks(errorUnlockAccount)('should show error when unlock account fai
...
+  const identityPage = await setup(t);
+  await identityPage.clickUnlockAccountLink();
+
+  const selectFactorPage = new SelectFactorPageObject(t);
+  await t.expect(selectFactorPage.getFormTitle()).eql('Unlock account?');
+  await t.expect(selectFactorPage.getFactorsCount()).eql(2);
+  await selectFactorPage.fillIdentifierField('username');
+  await selectFactorPage.selectFactorByIndex(0);
+
+  const challengeEmailPageObject =new ChallengeEmailPageObject(t);
```
[the remark this answers is not in the document]
Comment (vicentevillegas-okta) [reply test/testcafe/spec/UnlockAccountChallenge_spec.js]: done!

File: src/v2/view-builder/views/AutoRedirectView.js [hunk 3]
```diff
@@ -34,6 +36,18 @@ const Body = BaseForm.extend({
 
     return titleString;
   },
+  showMessages() {
+    if (this.isUnlockSucess()) {
+      const container = '.o-form-error-container';
```
Comment (justinabrokwah-okta) [inline src/v2/view-builder/views/AutoRedirectView.js:41]: Curious, do we have to use an error container for the success message?

File: src/v2/view-builder/views/AutoRedirectView.js [hunk 4]
```diff
@@ -34,6 +36,18 @@ const Body = BaseForm.extend({
...
+      const text = loc('oie.selfservice.unlock_user.landing.to.app.signing.in.message', 'login');
+      this.add(`<div class="ion-messages-container"><p>${text}</p></div>`, container);
+      return;
+    }
+    BaseForm.prototype.showMessages.call(this);
+  },
+  isUnlockSucess() {
```
Comment (justinabrokwah-okta) [inline src/v2/view-builder/views/AutoRedirectView.js:48]: nit: typo in the word Suc**c**ess

File: src/v2/view-builder/views/SelectAuthenticatorVerifyView.js [hunk 5]
```diff
@@ -14,12 +16,32 @@ export const Body = BaseForm.extend({
...
+      let options = {};
+      options.subtitle = loc('oie.selfservice.unlock_user.landing.to.app.success.message', 'login');
+      options.type = 'success';
+      options = createCallout(options);
+      BaseForm.prototype.showMessages.call(this, options);
+      return;
+    }
+    BaseForm.prototype.showMessages.call(this);
+  },
+  isUnlockSucess() {
```
Comment (justinabrokwah-okta) [inline src/v2/view-builder/views/SelectAuthenticatorVerifyView.js:42]: same nit here

**Review by justinabrokwah-okta — COMMENTED** [review]
PR and videos LGTM! Left a couple comments/nits

File: playground/mocks/config/responseConfig.js [hunk 6]
```diff
@@ -1053,5 +1054,7 @@ const selectOktaVerifyMethod = {
 };
 
 module.exports = {
-  mocks: idx
+  //mocks: idx
+  mocks: Test.UserUnlockAccountLandingApp.ChooseAuthAfterUnlock
+  //mocks: Test.UserUnlockAccountLandingApp.DirectLandingAfterUnlock
 };
```
[the remark this answers is not in the document]
Comment (vicentevillegas-okta) [reply playground/mocks/config/responseConfig.js:1060]: good catch @justinabrokwah-okta, I will undo the change for this part.

File: src/v2/view-builder/views/AutoRedirectView.js (hunk 3 above)
Comment (vicentevillegas-okta) [reply src/v2/view-builder/views/AutoRedirectView.js:41]: unfortunately yes, I tried with other containers and it didn't work

File: playground/mocks/config/responseConfig.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (vicentevillegas-okta) [reply playground/mocks/config/responseConfig.js]: suggestion applied!

File: src/v2/view-builder/views/AutoRedirectView.js (hunk 4 above)
Comment (vicentevillegas-okta) [reply src/v2/view-builder/views/AutoRedirectView.js]: suggestion applied!

File: src/v2/view-builder/views/AutoRedirectView.js (hunk 3 above)
Comment (justinabrokwah-okta) [reply src/v2/view-builder/views/AutoRedirectView.js:41]: Noted!

**Review by justinabrokwah-okta — APPROVED** [review]
Great work, Vicente! 🚀

File: packages/@okta/i18n/src/properties/login.properties [hunk 7]
```diff
@@ -1276,6 +1276,8 @@ oie.user.profile.division = Division
 oie.user.profile.department = Department
 oie.user.profile.managerId = Manager ID
 oie.user.profile.manager = Manager
+oie.selfservice.unlock_user.landing.to.app.success.message = Account successfully unlocked! Verify your account with a security method to continue.
+oie.selfservice.unlock_user.landing.to.app.signing.in.message = Account successfully unlocked!
```
Comment (vicentevillegas-okta) [inline packages/@okta/i18n/src/properties/login.properties:1280]: @IanHakes-okta do you mind to take a look please? This is similar to the one reviewed [here](https://github.com/okta/okta-core/pull/67284)

Comment (kumarabhinav-okta) [reply packages/@okta/i18n/src/properties/login.properties:1280]: @BarbaraTaylor-okta can you review this, given ian is away.

**Review by jaredperreault-okta — APPROVED** [review]
Just make sure the new `login.properties` values are reviewed before merging

File: src/v2/view-builder/views/SelectAuthenticatorVerifyView.js [hunk 8]
```diff
@@ -14,12 +16,32 @@ export const Body = BaseForm.extend({
...
   },
   isPasswordRecoveryFlow() {
     return this.options.appState.get('isPasswordRecovery');
   },
   noButtonBar: true,
+  showMessages() {
+    if (this.isUnlockSuccess()) {
+      let options = {};
+      options.subtitle = loc('oie.selfservice.unlock_user.landing.to.app.success.message', 'login');
+      options.type = 'success';
```
Comment (xifengfang-okta) [inline src/v2/view-builder/views/SelectAuthenticatorVerifyView.js:35]: IIRC, when Idx returns the message object, it has a `class` property(possible value: info/success/warning) which can be used to style callout. but I am not sure it is usable here.

Comment (justinabrokwah-okta) [reply src/v2/view-builder/views/SelectAuthenticatorVerifyView.js:35]: Good callout - iirc I believe when Vicente first started working on this, that was the first thing he tried but it wasn't working as intended

File: packages/@okta/i18n/src/properties/login.properties (hunk 7 above)
Comment (nikitaapraj-okta) [reply packages/@okta/i18n/src/properties/login.properties:1280]: Reviewing on behalf of Ian. Looks good to me.

**Review by nikitaapraj-okta — APPROVED** [review]

**Review by nikhilvenkatraman-okta — APPROVED** [review]
i18n changes look good

**Pull request merged**